### PR TITLE
CDC fix duplicate unassgined

### PR DESCRIPTION
**Pull request by abraverm**

### CODE REVIEW

File: pkg/controller/clusterpool/clusterpool_controller_test.go [hunk 1]
```diff
@@ -287,6 +287,64 @@ func TestReconcileClusterPool(t *testing.T) {
...
+				),
+			},
+			expectedTotalClusters:        0,
+			expectedObservedSize:         0,
+			expectedObservedReady:        0,
+			expectedInventoryValidStatus: corev1.ConditionTrue,
+			expectedCDCurrentStatus:      corev1.ConditionUnknown,
+			expectInventory:              true,
+			expectError:                  true,
+		},
```
Comment (2uasimojo) [inline pkg/controller/clusterpool/clusterpool_controller_test.go:347]: These two tests still pass with the fix reverted

Comment (abraverm) [reply pkg/controller/clusterpool/clusterpool_controller_test.go:347]: Right, this part adds tests that were missing for [Reverse function](https://github.com/openshift/hive/pull/1866/files#diff-bbf4d3cb24b8cac87951b84ca973747902c00d4a9d662648c8e760a37569581dL740) and address the change.

**Review by 2uasimojo — COMMENTED** [review]
/lgtm

File: pkg/controller/clusterpool/clusterpool_controller_test.go [hunk 2]
```diff
@@ -338,6 +338,64 @@ func TestReconcileClusterPool(t *testing.T) {
...
+					testcd.WithUnclaimedClusterPoolReference(testNamespace, testLeasePoolName),
+					testcd.WithCustomization("test-cdc-1"),
+					testcd.Running(),
+				),
+			},
+			expectedTotalClusters:        1,
+			expectedObservedSize:         1,
+			expectedObservedReady:        1,
+			expectedInventoryValidStatus: corev1.ConditionTrue,
+			expectInventory:              true,
```
Comment (2uasimojo) [inline pkg/controller/clusterpool/clusterpool_controller_test.go:361]: here

File: pkg/controller/clusterpool/clusterpool_controller_test.go [hunk 3]
```diff
@@ -338,6 +338,64 @@ func TestReconcileClusterPool(t *testing.T) {
...
+					testcdc.WithCD("c1"),
+					testcdc.Available(),
+				),
+			},
+			expectedTotalClusters:        0,
+			expectedObservedSize:         0,
+			expectedObservedReady:        0,
+			expectedInventoryValidStatus: corev1.ConditionTrue,
+			expectedCDCurrentStatus:      corev1.ConditionUnknown,
+			expectInventory:              true,
```
Comment (2uasimojo) [inline pkg/controller/clusterpool/clusterpool_controller_test.go:379]: here

File: pkg/controller/clusterpool/clusterpool_controller_test.go [hunk 4]
```diff
@@ -338,6 +338,64 @@ func TestReconcileClusterPool(t *testing.T) {
...
+					testcdc.WithPool("c1"),
+					testcdc.Available(),
+				),
+			},
+			expectedTotalClusters:        0,
+			expectedObservedSize:         0,
+			expectedObservedReady:        0,
+			expectedInventoryValidStatus: corev1.ConditionTrue,
+			expectedCDCurrentStatus:      corev1.ConditionUnknown,
+			expectInventory:              true,
```
Comment (2uasimojo) [inline pkg/controller/clusterpool/clusterpool_controller_test.go:396]: and here.

You removed this field in the [last PR](https://github.com/openshift/hive/pull/1855).

**Pull request merged**